### ISSUE TITLE
PHP Stan support

### ISSUE THREAD

**Issue by sandeshjangam** [lorisleiva/laravel-actions#260]
It doesn't throw errors in PHPStan if args type, numbers etc. mismatches...
```php
namespace App\Authentication\Actions;

use Lorisleiva\Actions\Concerns\AsAction;

class UpdateUserPassword
{
    use AsAction;

    public function handle(User $user, string $newPassword)
    {
        $user->password = Hash::make($newPassword);
        $user->save();
    }
}
```
```
// Equivalent to "UpdateUserPassword::make()->handle($user, 'secret')".
UpdateUserPassword::run($user, 'secret');
```

If I use this - UpdateUserPassword::make()->handle($user, 'secret')
Larstan properly check args type, count etc

For this - UpdateUserPassword::run($user, 'secret')
Larstand doesn't trigger any error 

How can I add support to this? I am open to creating a PR. A little head start will be good...

**Comment by sandeshjangam** [comment]
@lorisleiva  I am open to creating a PR. A little head start will be good...

**Comment by Wulfheart** [comment]
I don’t think that this can be achieved in this package due to the dynamic nature of the package. 

Did you try the ide-helper package yet?

**Comment by sandeshjangam** [comment]
> I don’t think that this can be achieved in this package due to the dynamic nature of the package.
> 
> Did you try the ide-helper package yet?

Nope, I haven't yet. I am not sure how the ide-helper package will help to get it compatible with Larstan(PHPStan)?

Can you please elaborate?

**Comment by Patabugen** [comment]
@sandeshjangam The ide-helper package generates docblocks (or other files/annotations) based on your actual code which phpStan can use to understand your code better.

For example, it might add a docblock bit which can tell phpStan the arguments for `UpdateUserPassword::run()`

There's an extension for ide-helper just for laravel-actions which might help:

https://github.com/Wulfheart/laravel-actions-ide-helper

**Comment by kylemilloy** [comment]
I think this is possible with generics, yes? Would really like to have this since everything getting set to "mixed" isn't terribly helpful.

**Comment by Wulfheart** [comment]
This topic definitely needs more love.

**Comment by DawidMazur** [comment]
I think this also could get fix:

![Image](https://github.com/user-attachments/assets/67fb1c79-1882-47e1-a2aa-20308b93af08)
